MvvmCross.DroidX/RecyclerView/MvxRecyclerAdapter.cs

```diff
@@ -142,7 +142,8 @@ public override AndroidX.RecyclerView.Widget.RecyclerView.ViewHolder OnCreateVie
 
         protected virtual View InflateViewForHolder(ViewGroup parent, int viewType, IMvxAndroidBindingContext bindingContext)
         {
-            return bindingContext.BindingInflate(viewType, parent, false);
+            var layoutId = ItemTemplateSelector.GetItemLayoutId (viewType);
```

Comment:
@alexshikov @Cheesebaron 

I might be missing something but isn't the viewType parameter here the actual layout resource id? Why would the GetItemLayoutId method need to be called again? 

On line 127, the exact same method is called except this time passing in the index of the object in the backing data source, which makes sense - the template selector can return the layout id based on the index of the item in the list.

Reply:
Because `viewType` is not the same as `layoutId`. Even though line 127 return `layoutId` you can override `GetItemViewType` and return your own data type (please see any Android examples with header/footer). And that means `InflateViewForHolder` should convert `viewType` to `layoutId` to pass it to `BindingInflate`.